File: modules/common/src/main/java/com/common/utils/PathNameBuilder.java
package com.common.utils;

import java.util.Arrays;

/**
 * Path name을 조합하기 위한 유틸 클래스
 */
@SuppressWarnings("unused")
public class PathNameBuilder {
	
	private PathNameBuilder() {
		throw new IllegalStateException("Utility class");
	}
	  
	public static String get(String...strings) {
		return get(false, strings);
	}
	
	public static String get(boolean isDirectory, String...strings) {
		StringBuilder stringBuilder = new StringBuilder();
		Arrays.stream(strings).forEach(string-> stringBuilder.append("/").append(string));
		if (isDirectory) {
			return stringBuilder.append("/").substring(1);
		} else {
			return stringBuilder.substring(1);
		}
	}
}

File: modules/common/src/main/kotlin/com/media/bus/common/utils/PathNameBuilder.kt
package com.media.bus.common.utils

/**
 * ## Path name을 조합하기 위한 유틸 클래스
 */
@Suppress("unused")
object PathNameBuilder {

    @JvmStatic
    fun get(vararg strings: String): String = get(false, *strings)

    @JvmStatic
    fun get(isDirectory: Boolean, vararg strings: String): String {
        val sb = StringBuilder()
        strings.forEach { s -> sb.append("/").append(s) }
        return if (isDirectory) {
            sb.append("/").substring(1)
        } else {
            sb.substring(1)
        }
    }
}
